spiketoolkit/postprocessing/features.py

```diff
@@ -96,149 +115,202 @@ def compute_unit_template_features(waveforms,
-    --------
-    repolarization_slope : slope of return to baseline (V / s)
-    """
-
-    max_point = np.argmax(np.abs(waveform))
-
-    waveform = - waveform * (np.sign(waveform[max_point]))  # invert if we're using the peak
-
-    repolarization_slope = linregress(timestamps[max_point:max_point + window], waveform[max_point:max_point + window])[
-        0]
+# def compute_unit_template_duration(waveform, timestamps):
+#     """
+#     Duration (in seconds) between peak and trough
+#     Inputs:
+#     ------
+#     waveform : numpy.ndarray (N samples)
+#     timestamps : numpy.ndarray (N samples)
+#     Outputs:
+#     --------
+#     duration : waveform duration in milliseconds
+#     """
+#
+#     trough_idx = np.argmin(waveform)
+#     peak_idx = np.argmax(waveform)
+#
+#     # to avoid detecting peak before trough
+#     if waveform[peak_idx] > np.abs(waveform[trough_idx]):
+#         duration = timestamps[peak_idx:][np.where(waveform[peak_idx:] == np.min(waveform[peak_idx:]))[0][0]] - \
+#                    timestamps[peak_idx]
+#     else:
+#         duration = timestamps[trough_idx:][np.where(waveform[trough_idx:] == np.max(waveform[trough_idx:]))[0][0]] - \
+#                    timestamps[trough_idx]
+#
+#     return duration * 1e3
 
-    return repolarization_slope * 1e-6
 
+def compute_trough_to_peak_duration(sorting,
+                                    unit_ids=None,
+                                    invert_template=False,
+                                    save_property_or_features=True,
+                                    return_through_and_peak_idx=False):
 
-def compute_unit_template_recovery_slopes(waveform, timestamps, window=20):
-    """
-    Spike recovery slope (after repolarization)
-    Inputs:
-    ------
-    waveform : numpy.ndarray (N samples)
-    timestamps : numpy.ndarray (N samples)
-    window : int
-        Window (in samples) for linear regression
-    Outputs:
-    --------
-    recovery_slope : slope of recovery period (V / s)
-    """
+    if isinstance(unit_ids, (int, np.integer)):
+        unit_ids = [unit_ids]
+    elif unit_ids is None:
+        unit_ids = sorting.get_unit_ids()
+    elif not isinstance(unit_ids, (list, np.ndarray)):
+        raise Exception("unit_ids should be an int or list")
 
-    max_point = np.argmax(np.abs(waveform))
+    assert 'template' in sorting.get_shared_unit_property_names(), 'add template to unit properties'
 
-    waveform = - waveform * (np.sign(waveform[max_point]))  # invert if we're using the peak
+    durations = []
+    peaks = []
+    throughs = []
 
-    peak_idx = np.argmax(waveform[max_point:]) + max_point
+    for uid in unit_ids:
+        templates = sorting.get_unit_property(uid, 'template')
+        unit_durations = []
+        unit_peaks = []
+        unit_throughs = []
 
-    recovery_slope = linregress(timestamps[peak_idx:peak_idx + window], waveform[peak_idx:peak_idx + window])[0]
+        for i, channel_template in enumerate(templates):
```

Comment:
You can do this without the loop with:
`trough_idx  = np.argmin(templates, axis=1)`